common/rpc/interceptor/namespace_count_limit_test.go

```diff
@@ -0,0 +1,220 @@
+				"GetWorkflowExecutionHistory": 1,
+			},
+			expectRateLimit: true,
+		},
+		{
+			name:               "non-long poll request",
+			request:            &workflowservice.GetWorkflowExecutionHistoryRequest{WaitNewEvent: false},
+			numBlockedRequests: 2,
+			perInstanceLimit:   2,
+			methodName:         "/temporal.api.workflowservice.v1.WorkflowService/GetWorkflowExecutionHistory",
+			tokens: map[string]int{
+				"GetWorkflowExecutionHistory": 1,
+			},
+			expectRateLimit: false,
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			tc.run(t)
+		})
+	}
+}
+
+// run the test case by simulating a bunch of blocked pollers, sending a final request, and verifying that it is either
+// rate limited or not.
+func (tc *testCase) run(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	handler := tc.createRequestHandler()
+	interceptor := tc.createInterceptor(ctrl)
+	// Spawn a bunch of blocked requests in the background.
+	tc.spawnBlockedRequests(handler, interceptor)
+
+	// With all the blocked requests in flight, send the final request and verify whether it is rate limited or not.
+	_, err := interceptor.Intercept(context.Background(), tc.request, &grpc.UnaryServerInfo{
+		FullMethod: tc.methodName,
+	}, noOpHandler)
+
+	if tc.expectRateLimit {
+		assert.ErrorContains(t, err, "namespace concurrent poller limit exceeded")
+	} else {
+		assert.NoError(t, err)
+	}
+
+	// Clean up by unblocking all the requests.
+	handler.Unblock()
+
+	for i := 0; i < tc.numBlockedRequests; i++ {
+		assert.NoError(t, <-handler.errs)
+	}
+}
+
+func (tc *testCase) createRequestHandler() *testRequestHandler {
+	return &testRequestHandler{
+		started: make(chan struct{}),
+		respond: make(chan struct{}),
+		errs:    make(chan error, tc.numBlockedRequests),
+	}
+}
+
+// spawnBlockedRequests sends a bunch of requests to the interceptor which will block until signaled.
+func (tc *testCase) spawnBlockedRequests(
+	handler *testRequestHandler,
+	interceptor *NamespaceCountLimitInterceptor,
+) {
+	for i := 0; i < tc.numBlockedRequests; i++ {
+		go func() {
+			_, err := interceptor.Intercept(context.Background(), tc.request, &grpc.UnaryServerInfo{
+				FullMethod: tc.methodName,
+			}, handler.Handle)
+			handler.errs <- err
+		}()
+	}
+
+	for i := 0; i < tc.numBlockedRequests; i++ {
+		<-handler.started
+	}
+}
+
+func (tc *testCase) createInterceptor(ctrl *gomock.Controller) *NamespaceCountLimitInterceptor {
+	registry := namespace.NewMockRegistry(ctrl)
+	registry.EXPECT().GetNamespace(gomock.Any()).Return(&namespace.Namespace{}, nil).AnyTimes()
+
+	logger := log.NewNoopLogger()
+	perInstanceCountLimit := func(ns string) int {
+		return tc.perInstanceLimit
+	}
+	interceptor := NewNamespaceCountLimitInterceptor(
+		registry,
+		logger,
+		perInstanceCountLimit,
+		tc.tokens,
+	)
+
+	return interceptor
+}
+
+// noOpHandler is a grpc.UnaryHandler which does nothing.
+func noOpHandler(context.Context, interface{}) (interface{}, error) {
```

Comment:
I think `noop` is fine
```suggestion
func noopHandler(context.Context, any) (any, error) {
```